metricbeat/docs/modules/awsfargate.asciidoc

```diff
@@ -0,0 +1,86 @@
+////
+This file is generated! See scripts/mage/docs_collector.go
+////
+
+[[metricbeat-module-awsfargate]]
+[role="xpack"]
+== AWS Fargate module
+
+experimental[]
+
+Amazon ECS on Fargate provides a method to retrieve various metadata, network
+metrics, and Docker stats about tasks and containers. This is referred to as the
+https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-metadata-endpoint-v4-fargate.html[task metadata endpoint]
+and this endpoint is available per container.
+
+The environment variable is injected by default into the containers of Amazon
+ECS tasks on Fargate that use platform version 1.4.0 or later and Amazon ECS
+tasks on Amazon EC2 that are running at least version 1.39.0 of the Amazon ECS
+container agent.
+
+The awsfargate module is a Metricbeat module which collects AWS fargate metrics
+from task metadata endpoint.
+
+[float]
+== Introduction to AWS ECS and Fargate
+Amazon Elastic Container Service (Amazon ECS) is a highly scalable, fast,
+container management service that makes it easy to run, stop, and manage
+containers. ECS has two launch types that can define how compute resources will
+be managed: ECS EC2 and ECS Fargate.
+
+* *ECS EC2*
+
+ECS EC2 launches containers that run on EC2 instances. Users have to manage EC2
```

Comment:
We could add a note about how one can monitor this kind of containers. I suspect this could be done by using `docker` module directly installed on the EC2 instances but need to completely verify that (access to docker socket etc). wdyt?

Reply:
Good point! I wonder if we should only put one or two sentences here for this because it is mainly focusing on fargate?

Reply:
Sounds good!